LEAF_Request_Portal/sources/Service.php

```diff
@@ -135,20 +135,16 @@ public function removeMember($groupID, $member)
 
             if ($res[0]['locallyManaged'] == 1)
             {
-                $vars = array(':userID' => $member,
-                        ':groupID' => $groupID, );
-                $res = $this->db->prepared_query('DELETE FROM service_chiefs WHERE userID=:userID AND serviceID=:groupID', $vars);
-
                 $this->dataActionLogger->logAction(\DataActions::DELETE,\LoggableTypes::SERVICE_CHIEF,[
                     new LogItem("service_chiefs","serviceID", $groupID, $this->getServiceName($groupID)),
                     new LogItem("service_chiefs", "userID", $member, $this->getEmployeeDisplay($member))
                 ]);
+
+                $res = $this->db->prepared_query('DELETE FROM service_chiefs WHERE userID=:userID AND serviceID=:groupID', $vars);
```

Comment:
When you are logging items, run the logging after the process is completed and checked to have completed.  If you don't, you could end up with log files that have records of things that actually failed.

Reply:
Yeah I noticed this, but when I run this LOG after the event the record is already deleted from the table so it can't log it properly, is there a better way?

Reply:
Resolving as I am not familiar enough with underlying code to know what has to run in what order or could be cached to log after delete.